server/util/perms/perms.go

```diff
@@ -99,6 +99,50 @@ func FromACL(acl *aclpb.ACL) (int, error) {
 	return p, nil
 }
 
+func AuthenticatedUser(ctx context.Context, env environment.Env) (interfaces.UserInfo, error) {
```

Comment:
this feels really similar to AuthenticatedUser in auth -- it even has the same name!

Why don't you make AuthorizeRead take a context and env below, and then call an un-exported version of this function from AuthorizeRead?

Reply:
I wanted to keep the signature of `AuthorizeRead` harmonious with the existing `AuthorizeWrite`, and I also thought it made sense for those funcs to take the authenticatedUser directly, since calling `AuthenticatedUser(ctx)` results in some mildly expensive parsing (not to mention, it started to feel like they were doing too much).

This `AuthenticatedUser` (to me) feels like an "overload" of the other AuthenticatedUser -- it gives you the same result, but it adds some extra utility in that it does the null check for you. That's why I thought it belonged here, in this utility package.

Reply:
Thanks for the explantion!

I understand wanting AuthorizeRead and AuthorizeWrite to be similar -- makes sense.

* I don't love the "overload" idea for a function like this but i'm fine with it for now. If it's an issue / causes confusion we can always change it later.